notebooks/Binary Classification with the UCI Credit Card Payment Default Dataset.ipynb

```diff
@@ -0,0 +1,599 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Train-test split\n",
+    "df_train, df_test, Y_train, Y_test, A_train, A_test, A_str_train, A_str_test = train_test_split(\n",
+    "    dataset.drop(columns=['SEX', 'default payment next month']), \n",
+    "    Y, \n",
+    "    A, \n",
+    "    A_str,\n",
+    "    test_size = 0.3, \n",
+    "    random_state=12345,\n",
+    "    stratify=Y)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Using a Fairness Unaware Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We train an out-of-the-box `lightgbm` model on the biased data and assess several disparity metrics. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "lgb_params = {\n",
+    "    'objective' : 'binary',\n",
+    "    'metric' : 'auc',\n",
+    "    'learning_rate': 0.03,\n",
+    "    'num_leaves' : 10,\n",
+    "    'max_depth' : 3\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = lgb.LGBMClassifier(**lgb_params)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model.fit(df_train, Y_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Scores on test set\n",
+    "test_scores = model.predict_proba(df_test)[:, 1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Train AUC\n",
+    "roc_auc_score(Y_train, model.predict_proba(df_train)[:, 1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Predictions (0 or 1) on test set\n",
+    "test_preds = (test_scores >= np.mean(Y_train)) * 1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# LightGBM feature importance \n",
+    "lgb.plot_importance(model, height=0.6, title=\"Features importance (LightGBM)\", importance_type=\"gain\", max_num_features=15) \n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We notice that the biased feature `LIMIT_BAL`, appears as the most important feature in this model although it has no predictive power for an entire demographic segment in the data. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Helper functions\n",
+    "def group_equalized_odds_diff(y_true, y_pred, group):\n",
```

Comment:
These will break when @MiroDudik 's metrics change goes in, I think.

Reply:
yeah... let's merge this PR before then.